src/odemis/driver/hamamatsurx.py

```diff
@@ -0,0 +1,2132 @@
+        Convert the format provided by RemoteEx.
+        :return: (tuple) current binning
+        """
+        _binning = self.parent.CamParamGet("Setup", "Binning")  # returns list of format e.g. [2 x 2]
+        # convert as .resolution VA need tuple instead of list
+        binning = int(_binning[0].split("x")[0].strip(" ")), int(_binning[0].split("x")[1].strip(" "))
+
+        return binning
+
+    def _setBinning(self, value):
+        """
+        Called when .binning VA is modified. It actually modifies the camera binning.
+        :parameter value: (2-tuple int) binning value to set
+        :return: current binning value
+        """
+        # ResolutionVA need tuple instead of list of format [2 x 2]
+        binning = "%s x %s" % (value[0], value[1])
+        self.parent.CamParamSet("Setup", "Binning", binning)
+
+        prev_binning, self._binning = self._binning, value
+
+        # adapt resolution
+        # TODO check if really necessary
+        change = (prev_binning[0] / self._binning[0],
+                  prev_binning[1] / self._binning[1])
+        old_resolution = self.resolution.value
+        new_res = (int(round(old_resolution[0] * change[0])),
+                   int(round(old_resolution[1] * change[1])))
+
+        # fit
+        self.resolution.value = new_res
+
+        self._metadata[model.MD_BINNING] = self._binning  # update MD
+
+        if self._spectrograph:
+            self._updateWavelengthList()  # update WavelengthList when changing binning
+
+        return self._binning
+
+    def _setResolution(self, _=None):
+        """
+        Sets the resolution VA.
+        So far the full field of view is always used. Therefore, resolution only changes with binning.
+        :return: current resolution value
+        """
+        # Note: we can keep it simple as long as we do not provide to change the sensor size yet...
+        resolution = self._shape[:2]
+        new_res = (int(resolution[0] // self._binning[0]),
+                    int(resolution[1] // self._binning[1]))  # floor division: not below zero
+
+        if self._spectrograph:
+            self._updateWavelengthList()  # update WavelengthList when changing binning
+
+        return new_res
+
+    def _getCamExpTimeRange(self):
+        """
+        Get min and max values for the camera exposure time.
+        Values are in order. First to fourth values see CamParamInfoEx.
+        :return: tuple containing min and max exposure time
+        """
+        min_value = self.parent.CamParamInfoEx("Live", "Exposure")[4]
+        max_value = self.parent.CamParamInfoEx("Live", "Exposure")[-1]
+
+        min_value_raw, min_unit = min_value.split(' ')[0:2]
+        max_value_raw, max_unit = max_value.split(' ')[0:2]
+
+        self.min_exp = self.parent.convertUnit2Time(min_value_raw, min_unit)
+        self.max_exp = self.parent.convertUnit2Time(max_value_raw, max_unit)
+
+        range = (self.min_exp, self.max_exp)
+
+        return range
```

Comment:
You can just write `return min_exp, max_exp`.
In particular, avoid naming variables `range`, as it's built-in keyword. (Python is not too picky about this, and here it works, but it can lead to surprises).